src/main/java/com/book/nest/security/CustomUserDetailService.java

```diff
@@ -0,0 +1,20 @@
+package com.book.nest.security;
+
+import com.book.nest.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByEmail(username)
+                .orElseThrow(() -> new RuntimeException("Wrong login or password"));
```

Comment:
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Wrong login or password"));
```
Or create custom Exception